### ISSUE TITLE
Executing a Spring boot Application class starts it in compatibility/bower mode

### ISSUE THREAD

**Issue by marcushellberg** [vaadin/flow#5931]
Vaadin 14, beta 3. Project created with `vaadin init --pre`.

*Issue:*
If I open the project in IntelliJ and hit run in the IDE, it runs the Spring Boot application and starts the project in BOWER mode. 

If I run the app with `mvn package spring-boot:run`, it runs correctly in npm mode. 

*Expected result:*
I would expect that starting the app in any valid way would produce the same end result and run the app in npm mode.

**Comment by caalador** [comment]
The problem is that if you run it from the IDE without executing the prepare-frontend step we do not know what mode might be expected and thus fallback on the compatibility mode.

This is due to the fact that there is no 100% certain way of distinguishing what mode a project should be run in just from looking around at the structure. 

So the decided way is that the default mode is always compatibility  mode and when using the new maven goals we actually write the mode data to the `flow-build-info.json` as at that point we are 99.9% certain of the mode that you are running in.

Also running the Application from Idea is not the same as running `mvn spring-boot:run` (note you don't need to package anymore) as spring-boot:run will execute the plugin goal `prepare-frontend`

**Comment by Peppe** [comment]
I feel we have taken some wrong turn on how the modes are chosen in Flow. The “proper” mode (npm) should IMO be the default as eventually everybody will either have started a new project with npm or are thinking of doing the migration steps. It would be great if the developer could type the mode somewhere to force it, without having to rely on non-100% sure detection mode.

**Comment by caalador** [comment]
The developer can always give the system/application property `vaadin.compatibiityMode=false`

**Comment by Peppe** [comment]
Just to learn: Can you give `vaadin.compatibilityMode=true` as well to force bower mode? 

How do I give an application property? Can I write it to my pom.xml?

**Comment by mstahv** [comment]
Could we run the prepare-frontend goal when people import the project?

**Comment by Artur-** [comment]
Setting compatibilityMode to false does not help as you then hit
```
java.lang.NullPointerException: null
	at elemental.json.impl.JreJsonFactory.parse(JreJsonFactory.java:60) ~[gwt-elemental-2.8.2.vaadin2.jar:2.8.2.vaadin2]
	at elemental.json.Json.parse(Json.java:53) ~[gwt-elemental-2.8.2.vaadin2.jar:2.8.2.vaadin2]
	at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.appendNpmBundle(BootstrapHandler.java:854) ~[flow-server-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
```

**Comment by pleku** [comment]
For an intermediate solution, we will force the npm mode on for the Spring starters with the configuration. This will be made possible by Spring add-on `12.0.1` release which has the fix for the configuration not being applied.

**Comment by pleku** [comment]
Fixed with #173
